src/rai/rai/agents/state_based.py

```diff
@@ -80,6 +81,25 @@ class Report(BaseModel):
     )
 
 
+def get_stored_artifacts(tool_call_id: str) -> List[Any]:
+    with open("artifact_database.pkl", "rb") as file:
+        artifact_database = pickle.load(file)
+        if tool_call_id in artifact_database:
+            return artifact_database[tool_call_id]
+    return []
+
+
+def store_artifacts(tool_call_id: str, artifacts: List[Any]):
+    with open("artifact_database.pkl", "rb") as file:
+        artifact_database = pickle.load(file)
+        if tool_call_id not in artifact_database:
+            artifact_database[tool_call_id] = artifacts
+        else:
+            artifact_database[tool_call_id].extend(artifacts)
+    with open("artifact_database.pkl", "wb") as file:
+        pickle.dump(artifact_database, file)
+
+
```

Comment:
This is just a POC solution of a very real problem. Currently, llm vendors do not support tool messages with images.